src/lib/shopping-event-conversions.js

```diff
@@ -10,50 +11,93 @@ import { getUserId } from './local-storage';
 
 export type EventToFptiInputMapping = (event : Object) => FptiInput;
 
-function getStoredUserIds() : Object {
+const getDefaultFPTIEvent = () => {
   const storedUserIds = getUserId();
-  if (storedUserIds) {
-    return {
-      shopperId: storedUserIds.userId,
-      merchantProvidedUserId: storedUserIds.merchantProvidedUserId
-    };
-  } else {
-    return {};
-  }
-}
 
-function convertShoppingEventToFptiInput(
-  config : Config,
-  event : Object,
-  eventType : EventType
-) : FptiInput {
-  if (!event.user) {
-    event.user = config.user;
-  }
+  const data = {
+    shopperId: storedUserIds ? storedUserIds.userId : undefined,
+    merchantProvidedUserId: storedUserIds ? storedUserIds.merchantProvidedUserId : undefined
+  };
+
+  return data;
+};
+
+function getDefaultFPTIEventFromShoppingEvent(config : Config,
+  shoppingEvent : Object,
+  eventType : EventType) : FptiInput {
 
-  const storedUserIds = getStoredUserIds();
+  const eventData = {
+    ...shoppingEvent,
+    user: shoppingEvent.user || config.user
+  };
 
   const data : FptiInput = {
     eventName: eventType,
     eventType,
-    eventData: JSON.stringify(event),
-    shopperId: storedUserIds.shopperId,
-    merchantProvidedUserId: storedUserIds.merchantProvidedUserId
+    eventData
   };
 
   return data;
 }
 
+/**
+ * Generic function that performs the default conversion for events
+ * by stringifying the event object
+ *
+ * @param config
+ * @param shoppingEvent
+ * @param eventType
+ * @returns {FptiInput}
+ */
+const getDefaultConvertor = (config : Config,
+  shoppingEvent : Object,
+  eventType : EventType) => {
+
+  const defaultConvertedEvent =
+      getDefaultFPTIEventFromShoppingEvent(config, shoppingEvent, eventType);
+
+  const data : FptiInput =
+      Object.assign(
+        getDefaultFPTIEvent(),
+        { ...defaultConvertedEvent,
+          eventData: JSON.stringify(defaultConvertedEvent.eventData) }
+      );
+
+  return data;
+};
+
+const getViewProductEventConvertor = (config : Config,
```

Comment:
it is a bit harder to read. all we're doing is update the payload with currency if one defined in config. you can create a copy of ProductView event with currency updated and then call getDefaultConvertor.